### PR TITLE
1130 ccs handle property listing events 2021 stylee

### PR DESCRIPTION
**Pull request by LukeLoze1**
# Motivation and Context
CCSPropertyListed has new InterviewRequired Flag.  It no longer should handle refusals, link qids, invalid etc

# What has changed
Changed class and some tests

# How to test?
mvn clean install
and run ATs from ticket

# Links
https://trello.com/c/81w9gOMb/1130-ccs-handle-property-listing-events-2021-stylee-5

### CODE REVIEW

File: src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java [hunk 1]
```diff
@@ -228,13 +227,12 @@ public void testCreateCCSCaseWithRefusalReceived() {
             });
 
     // When
-    Case actualCase = underTest.createCCSCase(caseId, sampleUnit, HARD_REFUSAL, false);
+    Case actualCase = underTest.createCCSCase(caseId, sampleUnit);
```
Comment (liamtoozer) [inline src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java:230]: If we're no longer dealing with refusals in CCS case events then I think this test can go?

File: src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java [hunk 2]
```diff
@@ -197,7 +196,7 @@ public void testCreateCCSCaseWithRefusalNotReceived() {
             });
 
     // When
-    Case actualCase = underTest.createCCSCase(caseId, sampleUnit, null, false);
+    Case actualCase = underTest.createCCSCase(caseId, sampleUnit);
```
Comment (liamtoozer) [inline src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java:199]: If we're no longer dealing with refusals in CCS case events then I think this test can go?

File: src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java [hunk 3]
```diff
@@ -37,36 +34,14 @@ public CCSPropertyListedService(
...
-    // always generate a new uac-qid pair even if linking existing pair, this is in case field
-    // worker has to visit address again and launch an EQ
     uacService.createUacQidLinkedToCCSCase(caze, ccsPropertyListedEvent.getEvent());
-    if (hasOneOrMoreQids) {
-      handleUacQidLinksForCase(ccsProperty.getUac(), caze);
-    } else {
-      sendActiveCCSCaseToField(caze);
-    }
+
+    sendCCSCaseToFieldIdInterviewRequired(caze, ccsProperty.isInterviewRequired());
```
Comment (liamtoozer) [inline src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java:44]: `sendCCSCaseToFieldIdInterviewRequired` -> `sendCCSCaseToFieldIfInterviewRequired`

File: src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java [hunk 4]
```diff
@@ -4,14 +4,11 @@
...
 import java.time.OffsetDateTime;
-import java.util.List;
 import org.springframework.stereotype.Service;
 import uk.gov.ons.census.casesvc.logging.EventLogger;
 import uk.gov.ons.census.casesvc.model.dto.*;
 import uk.gov.ons.census.casesvc.model.entity.Case;
 import uk.gov.ons.census.casesvc.model.entity.EventType;
-import uk.gov.ons.census.casesvc.model.entity.RefusalType;
-import uk.gov.ons.census.casesvc.model.entity.UacQidLink;
 import uk.gov.ons.census.casesvc.model.repository.UacQidLinkRepository;
```
Comment (liamtoozer) [inline src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java:12]: Unused import now?

File: src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java (hunk 2 above)
Comment (LukeLoze1) [reply src/test/java/uk/gov/ons/census/casesvc/service/CaseServiceTest.java]: One of the tests has now been renamed to not reference refusals, the other one has been deleted

**Review by ryangrundy7 — APPROVED** [review]
Looks good 👍 Just need to address liams comments

File: src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java [hunk 5]
```diff
@@ -78,22 +54,10 @@ public void processCCSPropertyListed(
...
 
-  private void handleUacQidLinksForCase(List<UacDTO> qids, Case caze) {
-    for (UacDTO qid : qids) {
-      addUacLinkForQidAndCase(qid.getQuestionnaireId(), caze);
-    }
-  }
-
-  private void sendActiveCCSCaseToField(Case caze) {
-    if (caze.getRefusalReceived() == null && !caze.isAddressInvalid()) {
+  private void sendCCSCaseToFieldIfInterviewRequired(Case caze, boolean interviewRequired) {
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java:57]: having a function just to look at a well-named boolean seems pointless

File: src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java [hunk 6]
```diff
@@ -31,42 +30,19 @@ public CCSPropertyListedService(
...
-    // always generate a new uac-qid pair even if linking existing pair, this is in case field
-    // worker has to visit address again and launch an EQ
     uacService.createUacQidLinkedToCCSCase(caze, ccsPropertyListedEvent.getEvent());
-    if (hasOneOrMoreQids) {
-      handleUacQidLinksForCase(ccsProperty.getUac(), caze);
-    } else {
-      sendActiveCCSCaseToField(caze);
-    }
+
+    sendCCSCaseToFieldIfInterviewRequired(caze, ccsProperty.isInterviewRequired());
```
Comment (NickGrantONS) [inline src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java:45]: having a function just to look at a well-named boolean seems pointless

File: src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java (hunk 5 above)
Comment (LukeLoze1) [reply src/main/java/uk/gov/ons/census/casesvc/service/CCSPropertyListedService.java]: yeah fair enough, just sort of left it there

**Review by liamtoozer — APPROVED** [review]

**Pull request merged**